### PR TITLE
ESLint: Migrate to flat config, isolate Prettier, fix errors

### PR DESCRIPTION
**Pull request by jens-ox**
Hi all!
This PR updates ESLint to the new [flat config format](https://eslint.org/docs/latest/use/configure/configuration-files). Additionally, the following things have been done:
- separate Prettier and ESLint (as [recommended by Prettier](https://prettier.io/docs/en/integrating-with-linters.html#notes))
- move to [`typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint)
- remove old ESLint configurations and plugins
- fix issues detected by new config
- update scripts in `package.json` to include Prettier in CI

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -100,3 +100,6 @@ lib/**/*
 
 # Ignore jetbrains files
 .idea
+
+# ESBuild metafile
+metafile*.json
```
Comment (pavelzw) [inline .gitignore:105]:
```suggestion
metafile*.json

```

File: README.md [hunk 2]
```diff
@@ -117,10 +117,10 @@ test:
     matrix:
       environment: [py311, py312]
   steps:
-  - uses: actions/checkout@v4
-  - uses: prefix-dev/setup-pixi@v0.8.1
-    with:
-      environments: ${{ matrix.environment }}
+    - uses: actions/checkout@v4
```
Comment (pavelzw) [inline README.md:120]: This is checked by prettier, right?

File: src/main.ts [hunk 3]
```diff
@@ -129,18 +131,16 @@ const run = async () => {
...
-      exit(1)
-    } else if (typeof error === 'string') {
-      core.setFailed(error)
-      exit(1)
-    }
+run().catch((error: unknown) => {
+  if (core.isDebug()) {
     throw error
-  })
-  .then(() => exit(0)) // workaround for https://github.com/actions/toolkit/issues/1578
```
Comment (pavelzw) [inline src/main.ts:146]: Please keep this line, see #113

Comment (jens-ox) [reply src/main.ts:146]: Ok, I'll also add it to `post.ts` then

File: README.md (hunk 2 above)
Comment (jens-ox) [reply README.md:120]: yes

**Review by pavelzw — APPROVED** [review]
Thanks!

**Pull request merged**